File: webflux-dialogue/src/testFixtures/java/com/study/webflux/rag/fixture/MemoryFixture.java
package com.study.webflux.rag.fixture;

import java.time.Instant;

import com.study.webflux.rag.domain.dialogue.model.ConversationSessionId;
import com.study.webflux.rag.domain.memory.model.Memory;
import com.study.webflux.rag.domain.memory.model.MemoryType;

public final class MemoryFixture {

	public static final String DEFAULT_CONTENT = "사용자는 러닝을 좋아한다";
	public static final float DEFAULT_IMPORTANCE = 0.8f;

	private MemoryFixture() {
	}

	public static Memory createExperiential() {
		return Memory.create(ConversationSessionFixture.createId(),
			MemoryType.EXPERIENTIAL,
			DEFAULT_CONTENT,
			DEFAULT_IMPORTANCE);
	}

	public static Memory createExperiential(ConversationSessionId sessionId) {
		return Memory.create(sessionId,
			MemoryType.EXPERIENTIAL,
			DEFAULT_CONTENT,
			DEFAULT_IMPORTANCE);
	}

	public static Memory createFactual() {
		return Memory.create(ConversationSessionFixture.createId(),
			MemoryType.FACTUAL,
			"사용자의 직업은 개발자다",
			0.9f);
	}

	public static Memory createFactual(ConversationSessionId sessionId) {
		return Memory.create(sessionId, MemoryType.FACTUAL, "사용자의 직업은 개발자다", 0.9f);
	}

	public static Memory createWithId(String id, MemoryType type) {
		return new Memory(id, ConversationSessionFixture.createId(), type, DEFAULT_CONTENT,
			DEFAULT_IMPORTANCE, Instant.now(), Instant.now(), 1);
	}

	public static Memory createWithId(String id, ConversationSessionId sessionId, MemoryType type) {
		return new Memory(id, sessionId, type, DEFAULT_CONTENT, DEFAULT_IMPORTANCE,
			Instant.now(), Instant.now(), 1);
	}
}

File: app-miyou/src/testFixtures/kotlin/com/miyou/app/fixture/MemoryFixture.kt
package com.miyou.app.fixture

import com.miyou.app.domain.dialogue.model.ConversationSessionId
import com.miyou.app.domain.memory.model.Memory
import com.miyou.app.domain.memory.model.MemoryType
import java.time.Instant

object MemoryFixture {
    const val DEFAULT_CONTENT = "사용자는 러닝을 좋아한다"
    const val DEFAULT_IMPORTANCE = 0.8f

    @JvmStatic
    fun createExperiential(): Memory =
        Memory.create(
            ConversationSessionFixture.createId(),
            MemoryType.EXPERIENTIAL,
            DEFAULT_CONTENT,
            DEFAULT_IMPORTANCE,
        )

    @JvmStatic
    fun createExperiential(sessionId: ConversationSessionId): Memory =
        Memory.create(sessionId, MemoryType.EXPERIENTIAL, DEFAULT_CONTENT, DEFAULT_IMPORTANCE)

    @JvmStatic
    fun createFactual(): Memory =
        Memory.create(
            ConversationSessionFixture.createId(),
            MemoryType.FACTUAL,
            "사용자의 직업은 개발자다",
            0.9f,
        )

    @JvmStatic
    fun createFactual(sessionId: ConversationSessionId): Memory =
        Memory.create(sessionId, MemoryType.FACTUAL, "사용자의 직업은 개발자다", 0.9f)

    @JvmStatic
    fun createWithId(
        id: String,
        type: MemoryType,
    ): Memory =
        Memory(
            id,
            ConversationSessionFixture.createId(),
            type,
            DEFAULT_CONTENT,
            DEFAULT_IMPORTANCE,
            Instant.now(),
            Instant.now(),
            1,
        )

    @JvmStatic
    fun createWithId(
        id: String,
        sessionId: ConversationSessionId,
        type: MemoryType,
    ): Memory =
        Memory(
            id,
            sessionId,
            type,
            DEFAULT_CONTENT,
            DEFAULT_IMPORTANCE,
            Instant.now(),
            Instant.now(),
            1,
        )
}
